erts/emulator/beam/jit/instr_common.cpp

```diff
@@ -617,18 +617,50 @@ void BeamModuleAssembler::emit_i_make_fun(const ArgVal &Fun,
     a.mov(getXRef(0), RET);
 }
 
-void BeamModuleAssembler::emit_get_list(const ArgVal &Src,
+void BeamModuleAssembler::emit_get_list(const x86::Gp src,
                                         const ArgVal &Hd,
                                         const ArgVal &Tl) {
-    mov_arg(ARG1, Src);
+    x86::Gp boxed_ptr = emit_ptr_val(src, src);
 
-    x86::Gp boxed_ptr = emit_ptr_val(ARG1, ARG1);
+    switch (ArgVal::register_relation(Hd, Tl)) {
+    case ArgVal::Relation::consecutive: {
+        comment("(moving head and tail together)");
+        x86::Mem dst_ptr = getArgRef(Hd);
+        x86::Mem src_ptr = getCARRef(boxed_ptr);
+        dst_ptr.setSize(16);
```

Comment:
We seem to change the size of memory operands pretty often now, maybe we should add a size overload to `getArgRef` and friends, similar to `getSchedulerRegRef`?